lte/gateway/c/core/oai/lib/n11/SmfServiceClient.cpp

```diff
@@ -61,6 +61,8 @@ SetSMSessionContext create_sm_pdu_session_v4(
   // Encode APU, storing apn value
   req_common->set_apn((char*) apn);
 
+  // UE IPv4 address set
+  req_common->set_ue_ipv4((char*) ipv4_addr);
```

Comment:
why there is no test for this case?

Reply:
I see the test did not check that field.

Not only that, but all the IPs used on this test are the same. So if for any reason we are mixing the ips we are not catching that/

Please when finding an issue like that, remember that it must trigger a change on unit test to catch the problem automatically next time


https://sourcegraph.com/-/editor?remote_url=git%40github.com%3Auri200%2Fmagma.git&branch=master&file=lte%2Fgateway%2Fc%2Fcore%2Foai%2Ftest%2Fn11%2Ftest_smf_service_client.cpp&editor=JetBrains&version=v1.2.1&utm_product_name=CLion&utm_product_version=2021.1.2&start_row=72&start_col=11&end_row=72&end_col=11

Reply:
Now we've modified from `redirect_server_address` to `ue_ipv4` for setting the UE IPv4 address in `common_context`. UT is already present for `redirect_server_address`
https://sourcegraph.com/github.com/uri200/magma/-/blob/lte/gateway/c/core/oai/test/n11/test_smf_service_client.cpp?utm_product_name=CLion&utm_product_version=CLion#L68:12

Reply:
I really dont understand. On this PR you are setting 
```
req_common->set_ue_ipv4((char*) ipv4_addr);
```

But on your test you are comparing with `RedirectServer` ?? how does this relate?
```
EXPECT_TRUE(
  | magma::lte::RedirectServer::IPV4 ==
  | rat_req->mutable_pdu_address()->redirect_address_type());


```

Reply:
Regarding your first query; test is written as per the functionalities, so, in the functions previously we are using `redirect_server_address` and UT as well for UE IPv4 address assignment.

Reply:
What functionality is that?  Please provide context to help me understand why Redirect is used there. What is the functionality that field has? Where is it used?

Reply:
Thanks @ganeshg87 .   I just needed that `set_redirect_address_type(` removed.  It took us lots of time and comments just for one line of code.